index.d.ts

```diff
@@ -36,7 +36,7 @@ export type FieldConfig<T, Required extends boolean = boolean> = {
     submitParser?(value: T): T
 }
 
-export type InferForm<T extends () => Record<string, Field<any>>> = {
+export type InferForm<T extends (...args: Array<any>) => Record<string, Field<any> | undefined>> = {
```

Comment:
why `...args: Array<any>`

Reply:
I don't think its a good change

Reply:
when we want to pass some argument to a function that is used to define our config it is needed

example
```ts
export const useFormConfig = (initialName: string) => {
 // ....
 }
```

Now InferForm will throw an error

```...args: Array<any>``` means that it can be 0 to Infinite number of params in this config function

Reply:
![image](https://github.com/codemaskinc/react-form-builder-v2/assets/48803618/e85bfd9a-ce5d-4f2d-a315-1a5827639169)
![image](https://github.com/codemaskinc/react-form-builder-v2/assets/48803618/63d29e2a-6cd3-44bc-8905-0017507e31db)

Reply:
please add comment why it is there